OpenRVS/classes/OpenMultiPlayerWidget.uc

```diff
@@ -308,22 +329,219 @@ function Refresh(bool bActivatedByUser)
 	if ( ( m_LanServers.m_ClientBeacon == none ) || ( !m_LanServers.m_ClientBeacon.IsA('OpenClientBeaconReceiver') ) )
 		return;
 
+	//1.5 - if initiated by user, clear the list of current queries and start fresh
+	if ( bActivatedByUser )
+		OpenQueries.remove(0,OpenQueries.length);
 	//0.9: get each server in the list, then query for more info
+	//1.5 - query ServerList instead
+	j = 0;
+	while ( j < ServerList.length )
+	{
+		i = 0;
+		bFound = false;
+		//1.5 - only one open query to a server at a time
+		while ( i < OpenQueries.length )
+		{
+			if ( OpenQueries[i] == ServerList[j].IP )//aleady have query open
+			{
+				bFound = true;
+				break;
+			}
+			i++;
+		}
+		if ( !bFound )
+		{
+			OpenQueries[OpenQueries.length] = ServerList[j].IP;
+			//1.5 - get rough ping
+			if ( Timer == none )
+			{
+				Timer = new class'OpenTimer';
+				Timer.ClockSource = GetLevel();
+			}
+			Timer.StartTimer(ServerList[j].IP);
+			OpenClientBeaconReceiver(m_GameService.m_ClientBeacon).QuerySingleServer(self,
+				Left(ServerList[j].IP,InStr(ServerList[j].IP,":")),
+				int(Mid(ServerList[j].IP,InStr(ServerList[j].IP,":")+1))+1000);
+		}
+		j++;
+		//CurServer = R6WindowListServerItem(CurServer.Next);
+	}
+}
+
+//0.8 written
+//1.3 added mod keyword locking
+//1.5 receive locked info from specific servers, not master list
+function ReceiveServerInfo(string sIP,coerce int iNumP,coerce int iMaxP,string sGMode,string sMapName,string sSvrName,string sModName,bool bSvrLocked)
+{
+	local R6WindowListServerItem CurServer;
+	local int i,iTime;
+/*
+	//0.8
+	//find the server in the list that we received info for, and update
 	CurServer = R6WindowListServerItem(m_ServerListBox.GetItemAtIndex(0));
 	while ( CurServer != none )
 	{
-		//1.5 - get rough ping
-		if ( Timer == none )
+		if ( CurServer.szIPAddr == sIP )
 		{
-			Timer = new class'OpenTimer';
-			Timer.ClockSource = GetEntryLevel();
+			CurServer.iMaxPlayers = iMaxP;
+			CurServer.iNumPlayers = iNumP;
+			CurServer.szName = sSvrName;
+			CurServer.szGameType = GetLevel().GetGameNameLocalization(sGMode);
+			CurServer.szMap = sMapName;
+			//1.3 - grey out version if not the right mod
+			if ( caps(class'Actor'.static.GetModMgr().m_pCurrentMod.m_szKeyWord) != caps(sModName) )
+				CurServer.bSameVersion = false;
+			else
+				CurServer.bSameVersion = true;
+			CurServer.bLocked = bSvrLocked;//1.5 added locked here
+			//1.5 add ping
+			i = Timer.EndTimer(CurServer.szIPAddr);
+			if ( i != -1 )
+				CurServer.iPing = i / 2;//divide by two for single-trip time - consistent with how Ubi used to measure
+			CurServer = none;//break the while loop
 		}
-		Timer.StartTimer(CurServer.szIPAddr);
-		OpenClientBeaconReceiver(m_GameService.m_ClientBeacon).QuerySingleServer(self,
-			Left(CurServer.szIPAddr,InStr(CurServer.szIPAddr,":")),
-			int(Mid(CurServer.szIPAddr,InStr(CurServer.szIPAddr,":")+1))+1000);
+		else
 		CurServer = R6WindowListServerItem(CurServer.Next);
+	}*/

```

Comment:
would remove this block also, we can get the commented code from earlier commits in github

Reply:
Roger.